pkg/playback/yaml_serializer.go

```diff
@@ -0,0 +1,138 @@
+package playback
+
+import (
+	"net/http"
+	"net/url"
+
+	"gopkg.in/yaml.v2"
+)
+
+// YAMLSerializer encodes/persists cassettes to files and decodes yaml cassette files.
+type YAMLSerializer struct{}
+
+// EncodeCassette takes in a playback.cassette and returns an []byte of the YAML-encoded cassette
+func (s YAMLSerializer) EncodeCassette(cassette Cassette) ([]byte, error) {
+	var encodedCassette YAMLCassette
+	for _, inter := range cassette {
+		yamlReq, err := s.serializeReq(inter.Request)
+		if err != nil {
+			return nil, err
+		}
+
+		yamlRes, err := s.serializeResp(inter.Response)
+		if err != nil {
+			return nil, err
+		}
+
+		encodedCassette = append(encodedCassette, YAMLInteraction{Type: inter.Type, Request: yamlReq.(YAMLRequest), Response: yamlRes.(YAMLResponse)})
+	}
+
+	return yaml.Marshal(encodedCassette)
+}
+
+// DecodeCassette takes in a []byte of YAML and returns a playback.cassette of it
+func (s YAMLSerializer) DecodeCassette(data []byte) (Cassette, error) {
+	yamlCassette := YAMLCassette{}
+	yaml.Unmarshal(data, &yamlCassette)
+
+	var decodedCassette Cassette
+	for _, inter := range yamlCassette {
+		decodedCassette = append(decodedCassette, interaction{
+			Type: inter.Type,
+			Request: httpRequest{
+				Headers: inter.Request.Headers,
+				Body:    []byte(inter.Request.Body),
+				Method:  inter.Request.Method,
+				URL:     inter.Request.URL,
+			},
+			Response: httpResponse{
+				Headers:    inter.Response.Headers,
+				Body:       []byte(inter.Response.Body),
+				StatusCode: inter.Response.StatusCode,
+			},
+		})
+	}
+
+	return decodedCassette, nil
+}
+
+// -------- SERIALIZATION is used to encode interfaces to YAML
+// serializeReq takes in an httpRequest and returns a YAMLRequest
+func (s YAMLSerializer) serializeReq(req httpRequest) (interface{}, error) {
+	return YAMLRequest{
+		req.Method,
+		string(req.Body),
+		req.Headers,
```

Comment:
One thing it occurs to me we need to do -- we need to redact the API key inside the authorization header. Maybe not in this PR.

Reply:
there is a pending task for this, but yeah not in this PR as it should prob not be at the serializer level